src/parser/clauses/index.h

```diff
@@ -0,0 +1,18 @@
+#ifndef _CLAUSE_INDEX_H
+#define _CLAUSE_INDEX_H
+
+typedef enum {
+  DROP_INDEX,
+  CREATE_INDEX
+} AST_IndexOpType;
+
+typedef struct {
+  const char *label;
+  const char *property;
+  AST_IndexOpType operation;
+} AST_IndexNode;
+
+AST_IndexNode* New_AST_IndexNode(const char *label, const char *property, AST_IndexOpType optype);
```

Comment:
Missing a clause validation function.

Reply:
What do you think should be checked? The only thing I can think of is that an index clause cannot coexist with any other clause, but I feel like our grammar will enforce this properly.